apps/callscreen/js/calls_handler.js

```diff
@@ -277,45 +277,47 @@ var CallsHandler = (function callsHandler() {
   }
 
   function handleCallWaiting(call) {
-    LazyL10n.get(function localized(_) {
-      var number = call.secondId ? call.secondId.number : call.id.number;
+    var number = call.secondId ? call.secondId.number : call.id.number;
+
+    if (!number) {
+      navigator.mozL10n.setAttributes(CallScreen.incomingNumber,
+                                      'withheld-number');
+      // Force a synchronous translation so we can resize the string
+      navigator.mozL10n.translateFragment(CallScreen.incomingNumber);
```

Comment:
I'm trying to address a similar issue in bug 1155901 so I hope that the fix for that will apply here too.